src/v2/components/CompactChannel/index.tsx

```diff
@@ -0,0 +1,173 @@
+  text-decoration: none;
+  break-inside: avoid;
+  overflow: hidden;
+  cursor: pointer;
+  ${boxMixin}
+  ${channelVisibilityForegroundColor}
+
+  ${Label} {
+    ${channelVisibilityForegroundColor}
+  }
+`
+
+const Button = styled(GenericButtonLink).attrs({ py: 3, px: 4, f: 0 })``
+
+export const EmptyCompactChannel = ({ children, ...props }) => (
+  <Container {...props}>
+    {typeof children === 'string' ? (
+      <Primary>
+        <Label f={4}>{children}</Label>
+      </Primary>
+    ) : (
+      children
+    )}
+  </Container>
+)
+
+interface CompactChannelProps {
+  channel: Channel
+  showEditButton?: boolean
+  refetchQueries?: PureQueryOptions[]
+}
+
+const openEditChannel = (
+  id: string | number,
+  refetchQueries: PureQueryOptions[],
+  client: ApolloClient<object>
+) => {
+  const modal = new Modal(ManageChannel, {
+    id,
+    refetchQueries,
+    onDelete: () => {
+      client.writeFragment({
+        id: `Channel:${id}`,
+        fragment: compactChannel,
+        data: null,
+      })
+      modal.close()
+    },
+    onUpdate: () => {
+      modal.close()
+    },
+  })
+  modal.open()
+}
+
+export const CompactChannel: React.FC<CompactChannelProps> = ({
+  channel,
+  showEditButton = false,
+  refetchQueries,
+  ...rest
+}) => {
+  const [mode, setMode] = useState<'resting' | 'hovered'>('resting')
+  const client = useApolloClient()
+
+  const handleMouseOver = useCallback(() => {
+    showEditButton && setMode('hovered')
+  }, [showEditButton, setMode])
+
+  const handleMouseOut = useCallback(() => {
+    showEditButton && setMode('resting')
+  }, [showEditButton, setMode])
+
+  const onEditClick = e => {
+    e.preventDefault()
+    e.stopPropagation()
+    openEditChannel(channel.id, refetchQueries, client)
+  }
+
+  return (
+    <Container
+      href={channel.href}
+      visibility={channel.visibility}
+      {...rest}
+      onMouseEnter={handleMouseOver}
+      onMouseLeave={handleMouseOut}
+    >
+      <Primary>
+        <Label f={4} dangerouslySetInnerHTML={{ __html: channel.title }} />
+
+        {channel.visibility === 'private' && <BorderedLock mr={3} />}
+
+        <Label f={1} ml={6} secondary>
+          <Count label="block" amount={channel.counts.contents} />
+        </Label>
+      </Primary>
+
+      {mode === 'resting' && (
+        <Label f={1} textAlign="right">
+          by {channel.owner.name}
+          {channel.owner.__typename === 'Group' && (
+            <GroupBadge
+              visibility={channel.owner.visibility}
+              color={`channel.${channel.visibility.toLowerCase()}`}
+            />
+          )}
+        </Label>
+      )}
+
+      {mode === 'hovered' && (
+        <Label f={1} textAlign="right">
+          <Button color={`channel.${channel.visibility}`} onClick={onEditClick}>
+            Edit
+          </Button>
+        </Label>
+      )}
+    </Container>
+  )
+}
+
+export const MemoizedCompactChannel = React.memo(CompactChannel)
```

Comment:
Why not just wrap compact channel directly, why the namespacing?

Reply:
mm sure, can do